src/query/functions/src/scalars/timestamp/src/datetime.rs

```diff
@@ -2362,6 +2365,167 @@ fn register_rounder_functions(registry: &mut FunctionRegistry) {
         }),
     );
 
+    registry
+        .register_passthrough_nullable_3_arg::<DateType, UInt64Type, StringType, DateType, _, _>(
+            "to_time_slice_of_year",
```

Comment:
Implementing a separate function for each type might be a bit complicated. We can consider implementing a `time_slice` function to simplify the code